src/urllib3/response.py

```diff
@@ -455,6 +457,13 @@ def data(self):
         if self._fp:
             return self.read(cache_content=True)
 
+    def json(self) -> Mapping[str, str]:
+        res = self.data
+        if isinstance(res, bytes):
+            return json.loads(res.decode("utf-8"))
+        else:
+            return json.loads(res)
```

Comment:
The `isinstance` check is not necessary since `self.data` is always bytes. (And `json.loads` does this already internally anyway.). This is enough:

```suggestion
        return json.loads(self.data)
```